packages/runtime/src/components/builtin/Text/EditableText.tsx

```diff
@@ -171,7 +174,22 @@ const EditableText = forwardRef(function EditableText(
     if (event.relatedTarget == null) editor.select(selection)
   }, [])
 
-  const isInBuilder = useIsInBuilder()
+  // If the user selects any text when they are in interaction mode or content mode,
+  // deselect any text when they go to the build mode.
+  useEffect(() => {
+    if (editMode === EditModes.BUILD) editor?.deselect().blur()
+  }, [editMode, editor])
```

Comment:
I don't think that we should have a code-path for this, and instead, switching between modes should blur the entire document. This is also related to how we handle exiting interaction mode to go back to build mode

Reply:
Yeah, something like this makes sense. We can make a middleware that handles focus and selection